ash/src/extensions/debug_utils.rs

```diff
@@ -0,0 +1,166 @@
+#![allow(dead_code)]
+use prelude::*;
+use std::ffi::CStr;
+use std::mem;
+use version::{DeviceV1_0, InstanceV1_0};
+use {vk, RawPtr};
+
+#[derive(Clone)]
+pub struct DebugUtils {
+    debug_utils_fn: vk::ExtDebugUtilsFn,
+}
+
+impl DebugUtils {
+    pub fn new<I: InstanceV1_0, D: DeviceV1_0>(
+        instance: &I,
+        device: &D,
+    ) -> Result<DebugUtils, Vec<&'static str>> {
+        let debug_utils_fn = vk::ExtDebugUtilsFn::load(|name| unsafe {
+            mem::transmute(instance.get_device_proc_addr(device.handle(), name.as_ptr()))
+        })?;
+        Ok(DebugUtils {
+            debug_utils_fn: debug_utils_fn,
+        })
+    }
+
+    pub fn name() -> &'static CStr {
+        CStr::from_bytes_with_nul(b"VK_EXT_debug_utils\0").expect("Wrong extension string")
+    }
+
+    pub unsafe fn debug_utils_set_object_name_ext<H: vk::Handle>(
+        &self,
+        device: vk::Device,
+        object: H,
+        name: &CStr,
```

Comment:
I think it might be more ergonomic if this would be a `name: &str` and we create the `CStr` inside the function.

Reply:
I'd go with `name: &str`. The function is not performance critical.

Reply:
That would be inconsistent with other parts of the API (e.g. extension names, layers) and require allocation, making the interface no longer strictly zero-cost, though the cost is probably not important. Are we sure that's desired?

I use a macro like this to make `CStr`s convenient to come up with:
```rust
#[macro_export]
macro_rules! cstr {
    ($str:expr) => { #[allow(unused_unsafe)] unsafe { ::std::ffi::CStr::from_bytes_with_nul_unchecked(concat!($str, "\0").as_bytes()) } }
}
```

Reply:
Ah right I didn't think about the null terminator, yeah `CStr` is probably fine.